### ISSUE TITLE
Upscale from PNG to JPG creates incorrect file extension *.jpg.png

### ISSUE THREAD

**Issue by nilei** [upscayl/upscayl#653]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#-faq).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.
- [ ] I have tried [setting my GPU ID](https://github.com/upscayl/upscayl/wiki/Guide#gpu-id).
- [ ] I am using Windows and I've tried [common troubleshooting steps](https://github.com/upscayl/upscayl/wiki/Troubleshooting).

### Describe the Bug

Upscaling a single PNG file to a JPG file as output format creates the new upscaled file with an incorrect file extension in the form of "*.jpg.png". The file can be renamed to "*.jpg" and opened with no issues.

### To Reproduce

1. Set output format to JPG
2. Upscale a single PNG file

### Version (or commit hash)

2.9.8

### Select your Operating System

MacOS

### OS Version

14.2.1

### GPU Name

M1

### Expected Behavior

The upscaled file should have a *.jpg file extension only.

### Paste the logs

```sh
too much personal and additional not relevant infos from previous upscale processes in logs. Don't know where to start copy and pasting.
```

**Comment by NayamAmarshe** [comment]
Found the issue.

Upscayl-ncnn, when it sees a PNG with alpha channel, it only outputs a PNG. That's why it appends a .png to the .jpg.

For the PNGs that do not have an alpha channel, this wouldn't happen.

It is a bug, you're right but for the time being, you can use PNG or WebP export for PNGs with an alpha channel. The other PNGs will work fine though.

**Comment by NayamAmarshe** [comment]
It's strange that it doesn't work because the logic seems right: 
https://github.com/upscayl/upscayl/blob/3c19a6c5a9cad4c9bf14e82c8c145d7cce792368/electron/commands/image-upscayl.ts#L171C6-L171C6

**Comment by NayamAmarshe** [comment]
Aha, got it! So it skips the JPG conversion for [4x scale and 0% compression](https://github.com/upscayl/upscayl/blob/3c19a6c5a9cad4c9bf14e82c8c145d7cce792368/electron/utils/convert-and-scale.ts#L15). Ah good edge-case! Thanks for reporting @nilei 

I suppose we should skip the conversion in case of Alpha channel too, which I didn't take into account.

For now, here's a simple fix: Set the compression to 1% :)

**Comment by nilei** [comment]
I think, the user will still wonder why the outcome is still PNG then. What is about replacing the alpha channel with full black? I am not an expert with this, but I know this behaviour from other software.

**Comment by NayamAmarshe** [comment]
> I think, the user will still wonder why the outcome is still PNG then. What is about replacing the alpha channel with full black? I am not an expert with this, but I know this behaviour from other software.

Try setting the compression to 1%, it should work.

For now, I can't release a new version just for this bug alone. I'll have to make some additional changes and bug-fixes.

**Comment by nilei** [comment]
That's okay. I do not expect you to make a release for a single bug. Also providing the workaround with 1% compression is fine.

I was just referring to the idea of skipping the conversion to JPG in case of an PNG with an alpha channel. I think it would be better to replace the alpha channel with a solid color or just black or white. This would result into JPG files as expected and as assumed when setting the output format to JPG. Explaining this to the user is much more convenient than explaining that the setting was not respected because of an alpha channel in the image. I hope you understand my concern. 🙂 

-- Cheers, Nils

**Comment by NayamAmarshe** [comment]
> That's okay. I do not expect you to make a release for a single bug. Also providing the workaround with 1% compression is fine.
> 
> I was just referring to the idea of skipping the conversion to JPG in case of an PNG with an alpha channel. I think it would be better to replace the alpha channel with a solid color or just black or white. This would result into JPG files as expected and as assumed when setting the output format to JPG. Explaining this to the user is much more convenient than explaining that the setting was not respected because of an alpha channel in the image. I hope you understand my concern. 🙂 
> 
> -- Cheers, Nils 
> 

Oh yes, absolutely. We actually already replace alpha channels with the black color. It's just that this bug is skipping that process due to the flawed logic in code :)

Thanks a lot for reporting this though, It'll definitely be fixed in v2.9.9 and I'll probably also add a color picker for alpha channel replacement.
